src/Nethermind/Nethermind.Evm/EvmState.cs

```diff
@@ -213,6 +213,35 @@ static void ThrowIsInUse()
         }
     }
 
+    /// <summary>
+    /// Constructor for a frame <see cref="EvmState"/> beneath top level.
+    /// </summary>
+    public static EvmState RentTopLevelFrame(
+        long gasAvailable,
+        long outputDestination,
+        long outputLength,
+        ExecutionType executionType,
+        bool isStatic,
+        bool isCreateOnPreExistingAccount,
+        in ExecutionEnvironment env,
+        in StackAccessTracker stateForAccessLists,
+        in Snapshot snapshot)
+    {
+        EvmState state = Rent();
+        state.Initialize(
+            gasAvailable,
+            outputDestination,
+            outputLength,
+            executionType,
+            isTopLevel: true,
```

Comment:
So is it top level or not? Comment says "beneath top level"

Reply:
it is not technically top level, but it acts like top level.
we need to terminate the execution when we are done executing this frame

Reply:
Agreed